clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -2074,6 +2087,7 @@ class AnnotatingParser {
       if (ctx.ContextType == Context::StructArrayInitializer)
         return LT_ArrayOfStructInitializer;
 
+    markTokenAsTemplateArgumentInLine();
```

Comment:
I think you want this before the preceding loop. And can you get it into the first pass of the line? Just going through all tokens again doesn't seem nice.